docs/workflows/workflows.md

```diff
@@ -0,0 +1,195 @@
+  service that can be used to run predictions against the model.
+inputs:
+  - name: mlflow-codeset
+    description: an MLFlow compatible codeset
+    type: codeset
+  - name: predictor
+    description: type of predictor engine
+    type: string
+    default: auto
+outputs:
+  - name: prediction-url
+    description: "The URL where the exposed prediction service endpoint can be contacted to run predictions."
+    type: string
+steps:
+  - name: builder
+    image: ghcr.io/fuseml/mlflow-builder:v0.3.0
+    inputs:
+      - name: mlflow-codeset
+        codeset:
+          name: "{{ inputs.mlflow-codeset }}"
+          path: /project
+    outputs:
+      - name: image
+  - name: trainer
+    image: "{{ steps.builder.outputs.image }}"
+    inputs:
+      - name: mlflow-codeset
+        codeset:
+          name: "{{ inputs.mlflow-codeset }}"
+          path: "/project"
+    outputs:
+      - name: mlflow-model-url
+    extensions:
+      - name: mlflow-tracking
+        product: mlflow
+        service_resource: mlflow-tracking
+      - name: mlflow-store
+        product: mlflow
+        service_resource: s3
+  - name: predictor
+    image: ghcr.io/fuseml/kserve-predictor:v0.3.0
+    inputs:
+      - name: model
+        value: "{{ steps.trainer.outputs.mlflow-model-url }}"
+      - name: predictor
+        value: "{{ inputs.predictor }}"
+      - name: mlflow-codeset
+        codeset:
+          name: "{{ inputs.mlflow-codeset }}"
+          path: "/project"
+    outputs:
+      - name: prediction-url
+    extensions:
+      - name: s3-storage
+        service_resource: s3
+      - name: kserve
+        service_resource: kserve-api
+```
+
+Let's analyze the workflow definition section by section and explore what it all means.
+
+
+```yaml
+name: mlflow-e2e
+description: |
+  End-to-end pipeline template that takes in an MLFlow compatible codeset,
+  runs the MLFlow project to train a model, then creates a KServe prediction
+  service that can be used to run predictions against the model.
+```
+
+The workflow `name` is used to identify the workflow in all subsequent workflow operations, such as assigning codesets to it or listing its executions. The `description` is used to describe the workflow.
+
+```yaml
+inputs:
+  - name: mlflow-codeset
+    description: an MLFlow compatible codeset
+    type: codeset
+  - name: predictor
+    description: type of predictor engine
+    type: string
+    default: auto
+outputs:
+  - name: prediction-url
+    description: "The URL where the exposed prediction service endpoint can be contacted to run predictions."
+    type: string
+```
+
+This is where the workflow's global parameters, inputs and outputs are declared. Currently, the only type of supported input for FuseML workflows and steps is the codeset. In this example:
+
+- the `mlflow-codeset` input is a codeset that must contain an MLflow project with python code that will be used to train a ML model
```

Comment:
It might be a good idea to add a link to the description of codesets

Reply:
done (I hope)

Reply:
Hm, I do not see any reference there. I had in mind a reference to codeset definition, but maybe we do not have one...

Reply:
no, we don't have a section dedicated to codesets :-(